firebase-firestore/src/test/java/com/google/firebase/firestore/remote/BloomFilterTest.java

```diff
@@ -0,0 +1,178 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.remote;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.stream.Stream;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE)
+public class BloomFilterTest {
+
+  @Test
+  public void testEmptyBloomFilter() {
+    BloomFilter bloomFilter = new BloomFilter(new byte[0], 0, 0);
+    assertTrue(bloomFilter.isEmpty());
+  }
+
+  @Test
+  public void testEmptyBloomFilterThrowException() {
```

Comment:
An another example, a better name for "testEmptyBloomFilterThrowException" is "constructorShouldThrowIAEOnNegativeHashValue". Also, this test case should be split up so that each test case tests only one thing. The tests for different invalid arguments should be in different test methods.

go/unit-testing-practices#behavior-testing-examples